### PR TITLE
refactor: updated frontend-build to v12 & resolved eslint issues

### PR DESCRIPTION
**Pull request by BilalQamar95**
**Ticket:**
[42: Upgrade eslint to v8.x](https://github.com/openedx/frontend-wg/issues/42)

**What changed?**
- Updated `frontend-build `to v12 (Eslint was updated in `frontend-build` version resulting in it's version being bumped to v12. This PR updates `frontend-build` to reciprocate eslint version update)
- Resolved `eslint` issues post update (Some methods were removed to satisfy `react/no-unused-class-component-methods` rule)

### CODE REVIEW

**Comment by mphilbrick211** [comment]
Hi @BilalQamar95! Just checking in on this PR - are you planning to run the tests?

**Comment by BilalQamar95** [comment]
> Hi @BilalQamar95! Just checking in on this PR - are you planning to run the tests?

Hi @mphilbrick211, I have updated the PR to latest frontend-build version & resolved subsequent `eslint` isues. PR is ready to be reviewed from my side.

**Comment by mphilbrick211** [comment]
Tagging @colinbrash for review!

**Comment by mphilbrick211** [comment]
Hi @BilalQamar95! Just checking in on this to see if you plan to pursue this PR - if so, it looks like some checks will need to be re-run, and branch conflicts will need to be resolved. Thanks!

**Comment by BilalQamar95** [comment]
@mphilbrick211 I have resolved the conflicts & updated the PR

**Comment by BilalQamar95** [comment]
Hi @mphilbrick211, I have resolved merge conflicts. As for pending tests I can't seem to figure out a way to run these, maybe the owning squad could shed some light on it.

**Comment by mphilbrick211** [comment]
Hi @colinbrash and @openedx/revenue-squad - could you please take a look into why the tests aren't running?

**Comment by pshiu** [comment]
Thanks for coordinating, @mphilbrick211!

The `node_js CI / tests (16) (pull_request)` check did not run because this PR changes the check to be called `node_js CI / tests (16, 8.5.x) (pull_request)` (and now supports specifying the Node _and_ NPM version).

I removed the old check as required in this repo's GitHub master branch protections.

**Comment by pshiu** [comment]
Review of release notes for @edx/frontend-build, ^9.1.3 → ^12.0.0:
* https://github.com/openedx/frontend-build/releases/tag/v10.0.0
  * Add linting of React Hooks.
* https://github.com/openedx/frontend-build/releases/tag/v11.0.0
  * Dropped Node 12 support.
* https://github.com/openedx/frontend-build/releases/tag/v12.0.0
  * Upgrades ESlint. May have more lint issues.

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
   },
   "devDependencies": {
     "@edx/browserslist-config": "^1.1.0",
-    "@edx/frontend-build": "^12.0.0",
+    "@edx/frontend-build": "12.0.3",
```
Comment (julianajlk) [inline package.json]: Curious to know the decision to pin this version versus the caret range?
We don't have auto merges for dependencies upgrades in this repo with Renovate.

Comment (BilalQamar95) [reply package.json]: I was doing frontend-build upgrade for all the mfes, pinning frontend-build version was based on feedback I received on some other PRs. For this repo I don't see an issue using caret range.

**Comment by arbrandes** [comment]
@julianajlk, @pshiu might we get an approval, here?

@BilalQamar95, in the meantime, can you merge the latest from master, here?

**Review by julianajlk — APPROVED** [review]

**Comment by arbrandes** [comment]
Looks like a security vulnerability in a dependency.  We should handle that separately.  Squashing and merging.  Thanks @BilalQamar95!

**Pull request merged**